rust/unicode/src/lib.rs

```diff
@@ -85,13 +97,15 @@ impl<'a> Iterator for LineBreakIterator<'a> {
                 Ordering::Equal => {
                     // LB3, break at EOT
                     self.ix += 1;
-                    return Some((self.s.len(), true));
+                    let i = (self.state as usize) * N_LINEBREAK_CATEGORIES;
+                    let new = LINEBREAK_STATE_MACHINE[i];
+                    return Some((self.s.len(), new >= 0xc0));
```

Comment:
although not part of this PR, I also dislike that this iterator returns a tuple; again I think a struct with named fields would make for a more ergonomic API.